File: merge_two_sorted_arrays.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if(n==0) return;
        int last  = n-1;
        if(m==0){
            for(int i=last; i>=0; i--){
                nums1[i]= nums2[last];
                last--;
            }
        return;
        }
        int first = m-1;
        
        int flag=0;
        for(int i=nums1.length-1;i>=0;i--){
            if(nums2[last] > nums1[first]){
                nums1[i]=nums2[last];
                last--;
            }else if(nums2[last] < nums1[first]){
                nums1[i]=nums1[first];
                first--;
            }
            else{
                nums1[i]=nums1[first];
                first--;
            }
            if(last <0) break;
            if(first <0 ) {flag=1; break;}
            
           
        }
        
         if(flag==1 ){
                for(int i=last; i>=0; i--){
                    nums1[i]=nums2[last];
                    last--;
                }
         }
        
    }
}

File: mergeTwoSortedArrays.py
"""
Time Complexity : O(m+n)
Space Complexity : O(1)
Did this code successfully run on Leetcode : Yes
Any problem you faced while coding this : none
"""

class Solution:
    def merge(self, nums1, m, nums2, n):
        """
        Do not return anything, modify nums1 in-place instead.
        """
        
        p1 = m - 1
        p2 = n - 1
        idx = m + n - 1
        
        while p1 >= 0 and p2 >= 0:
            if nums1[p1] > nums2[p2]:
                nums1[idx] = nums1[p1]
                p1 -= 1
            else:
                nums1[idx] = nums2[p2]
                p2 -= 1
            idx -= 1
            
        while p2 >= 0:
            nums1[idx] = nums2[p2]
            p2 -= 1
            idx -= 1
            
s = Solution()
nums1 = [4,5,8,0,0,0]
nums2 = [2,6,9]
s.merge(nums1, 3, nums2, 3)
print(nums1)